slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/ValidatorsDao.java

```diff
@@ -38,10 +38,21 @@ public Validator insertIfNotExist(final Handle handle, final Bytes validator) {
   }
 
   public List<Validator> registerValidators(final Handle handle, final List<Bytes> validators) {
-    final PreparedBatch batch =
-        handle.prepareBatch("INSERT INTO validators (public_key) VALUES (?)");
-    validators.forEach(b -> batch.bind(0, b).add());
-    return batch.executeAndReturnGeneratedKeys().mapToBean(Validator.class).list();
+    // adapted from https://stackoverflow.com/a/66704110/535610
+    return handle
+        .createQuery(
+            String.format(
+                "SELECT v_id as id, v_public_key as public_key FROM upsert_validators(array[%s])",
```

Comment:
Does it make sense for insertIfNotExist to also use the upsert_validators function?

Reply:
yes, I initially had it planned but it slipped my mind, removed insertIfNotExist method in favor of registerValidators.